From d7c7354c4857ed9934086e3bdaf2b07e55f46bd1 Mon Sep 17 00:00:00 2001
From: Simon Richter <Simon.Richter@hogyros.de>
Date: Mon, 13 May 2013 12:43:42 +0200
Subject: [PATCH 06/15] Add clGetDeviceInfo(..., CL_BUILT_IN_KERNELS, ...)
To: beignet@lists.freedesktop.org

Currently, there are no built-in kernels, so this function returns an empty
string.
---
 src/cl_device_id.c |    1 +
 src/cl_device_id.h |    2 ++
 src/cl_gt_device.h |    1 +
 3 files changed, 4 insertions(+)

Index: beignet-0.1+git20130514+19e9c58/src/cl_device_id.c
===================================================================
--- beignet-0.1+git20130514+19e9c58.orig/src/cl_device_id.c	2013-05-08 11:55:52.000000000 +0200
+++ beignet-0.1+git20130514+19e9c58/src/cl_device_id.c	2013-05-14 20:09:06.922022404 +0200
@@ -230,6 +230,7 @@
     DECL_STRING_FIELD(PROFILE, profile)
     DECL_STRING_FIELD(OPENCL_C_VERSION, opencl_c_version)
     DECL_STRING_FIELD(EXTENSIONS, extensions);
+    DECL_STRING_FIELD(BUILT_IN_KERNELS, built_in_kernels)
 
     case CL_DRIVER_VERSION:
       if (param_value_size_ret) {
Index: beignet-0.1+git20130514+19e9c58/src/cl_device_id.h
===================================================================
--- beignet-0.1+git20130514+19e9c58.orig/src/cl_device_id.h	2013-05-08 11:55:52.000000000 +0200
+++ beignet-0.1+git20130514+19e9c58/src/cl_device_id.h	2013-05-14 20:09:06.922022404 +0200
@@ -84,6 +84,7 @@
   const char *opencl_c_version;
   const char *extensions;
   const char *driver_version;
+  const char *built_in_kernels;
   size_t name_sz;
   size_t vendor_sz;
   size_t version_sz;
@@ -91,6 +92,7 @@
   size_t opencl_c_version_sz;
   size_t extensions_sz;
   size_t driver_version_sz;
+  size_t built_in_kernels_sz;
   /* Kernel specific info that we're assigning statically */
   size_t wg_sz;
   size_t compile_wg_sz[3];
Index: beignet-0.1+git20130514+19e9c58/src/cl_gt_device.h
===================================================================
--- beignet-0.1+git20130514+19e9c58.orig/src/cl_gt_device.h	2013-05-08 11:55:52.000000000 +0200
+++ beignet-0.1+git20130514+19e9c58/src/cl_gt_device.h	2013-05-14 20:09:06.922022404 +0200
@@ -72,6 +72,7 @@
 DECL_INFO_STRING(profile, "FULL_PROFILE")
 DECL_INFO_STRING(opencl_c_version, "OpenCL 1.10")
 DECL_INFO_STRING(extensions, "")
+DECL_INFO_STRING(built_in_kernels, "")
 DECL_INFO_STRING(driver_version, LIBCL_VERSION_STRING)
 #undef DECL_INFO_STRING
 
